### PR TITLE
Breathing changes

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

According to math done by Timberpoes and myself, air on a 1x1 square will last you ~3 hours, a emergency air tank ~17 minutes, and the large tanks 17 hours.

With the new changes a turf will last about 45 minutes, an emergency tank ~13 minutes and a large tank 4.25 hours

Plasmaman tanks will last the same amount of time as before

## Why It's Good For The Game

As it stands our rounds are too short for breathing atmos to really make a difference and this opens new interactions and generally makes it matter more.

Also opens the door towards making atmosia matter more beyond "The antag/fusion subdepartment"

## Changelog
:cl:
balance: Spacemen will now take 4x larger breaths
balance: Various pocket and belt-sized internals tanks have had their capacities increased 3x to 4x to account for the new breathing changes
/:cl:

### CODE REVIEW

**Comment by Ghilker** [comment]
finally atmos techs have a job!

File: code/game/objects/items/tanks/tank_types.dm
Comment (LemonInTheDark) [inline code/game/objects/items/tanks/tank_types.dm:147]: what would this be without the change?

**Comment by zxaber** [comment]
Doesn't go far enough if you ask me. With no vents running, a room with a human shouldn't last more than like ten minutes.

File: code/__DEFINES/atmospherics.dm
Comment (Timberpoes) [inline code/__DEFINES/atmospherics.dm:43]: After our discussions on this, I feel we might as well expand to explain why it's 8 seconds. That fact seemed to draw some confusion even from other coders.

```suggestion
/// liters in a normal breath. note that breaths are taken once every 4 life ticks, which is 8 seconds
```

**Review by Timberpoes — CHANGES_REQUESTED** [review]

**Comment by MrDoomBringer** [comment]
Totally agree with @zxaber. I feel like every time-to-breathe stat in the pr desc could be divided by 10 without any particularly crazy effects

File: code/game/objects/items/tanks/tank_types.dm
Comment (Timberpoes) [inline code/game/objects/items/tanks/tank_types.dm:163]: Breath volume increased x4, tank capacity increased x3. Is this comment still correct?

**Comment by Timberpoes** [comment]
@zxaber @MrDoomBringer 

Be prepared for a bit of a wall of text explaining the problem with making breathing air matter to atmos.

## Why air doesn't matter
On the topic of how little air matters, we hashed over it a lot in the Manuel Discord's coding sub-channel.

The issue we have is that we're a touch too close to life with our atmos sim. A turf has a volume of 2500L, which implies a turf is 2.5m^3. It turns out that a human sucks in relatively little oxygen in real life.

### Current live behaviour

In-code, a breath outside of internals is a 0.5L chunk of the turf's air mixture. This is a realistic value for a breath. But a breath every 8 seconds is not realistic.
If we simplify conditions to just O2, CO2 and N2 - As long as that breath has <10% CO2 and >16% O2, everything is fine. This is basically similar to real life too, these numbers pop up with a bit of Google-fu.
We replace the O2 with CO2 in a 1:1 ratio and that air mixture goes back to the turf. This is a bit more severe than real life, where some of the N2 is lost and some of the O2 does not become CO2.

Simulations indicate that we get 1360 breaths from a pristine 2500L roundstart turf filled with crisp, clean O2/N2 mix at 21:79 ratio.

image

At breath 1361, O2 is below 16% and we'll start suffering the effects of hypoxia. CO2 is nowhere near the necessary 10% to become an issue.

1361 breaths, one breath every 4 life ticks or 8 seconds, 10888 seconds, or a tad over 3 hours. Per turf.

In a 10x10 room, there are 100 turfs which is enough to sustain a single person for 300 hours. Or the entirety of all players across all /tg/ servers at peak time for a full 1.5 hour round with plenty of room to spare.

### Breaking away from realism entirely

To bring that level down to ANYTHING approaching dangerous, we can use a 200L turf which would imply a turf is 0.2m^3 or 20 cubic centimeters.
A 200L turf would reach dangerous levels after 109 breaths which is about 14 and a half minutes.
In this case, our 10x10 room would sustain a single person for a mere 24 hours.
It would sustain a station of 60 people, in that one 10x10 room, for 24 minutes.

### The problems?

The station itself contains many, many thousands of turfs and players are spread out across the station at almost all times. In reality, even with the hugely unrealistic 200L volume of air per turf you could turn air distro off roundstart and players wouldn't even notice over your average 90 minute round.

The underlying issue is that every new shift is a reset on the world. Every crew starts in a pristine station complete with full atmos tanks and a fresh helping of crisp, clean air. The reason real space stations have issues with atmospherics is because they don't bluespace-undo every 2 hours and actually have persistance over time.

We can't exactly make the shift start with stale air either. We must consider the affect on silicon players; especially AIs. Would you particularly enjoy your every round start being told to set every air alarm on the station to cycle nice, new, clean air in and scrub any contaminants and trace atmos gasses away?

Maybe you would, but others would not. And they have feelings too.

### What can we do after this PR gets merged?

Well, we take the above numbers and divide them by 4, really. That becomes the new norm. Air gets used 4x as fast by just breathing, so tens of hours become slightly less tens of hours.

You're still talking about atmos having a negligible impact on the station even with the uber unrealistic 200L volume, 20cm^3 turf sizes. Although you may have fun when 6 people clamber into a 2x1 emergency shuttle and end up asphyxiating before they reach CentCom.

## Conclusions?

This would have to be an area where realism gives way very heavily to game design in order to make breathing air actually matter. Our atmos sim has too much likeness to the real world.

I am not personally against okaying breaking with reality for the sake of gameplay, but the scope of such a PR would go well, well beyond this change.

File: code/game/objects/items/tanks/tank_types.dm
Comment (Timberpoes) [reply code/game/objects/items/tanks/tank_types.dm:147]: I get the feeling this comment in the code wasn't stating the old tanks got 21 breaths, but was emphasising how ridiculously small an e-tank is IRL. 21 breaths. That's a minute at best IRL and 3 minutes in-game with our Olypmian atheles who only breathe every 8 seconds. Our old e-tanks lasted much longer than that.

**Comment by LemonInTheDark** [comment]
Just to be sure, can you test these numbers in game? I wanna be sure they're accurate

**Comment by optimumtact** [comment]
image

**Comment by Timberpoes** [comment]
> Just to be sure, can you test these numbers in game? I wanna be sure they're accurate

@TiviPlus Did you take down any numbers from your in-game testing? I know you spawned like 100 moths in a GBJ and it took them basically forever to start asphyxiating for what was effectively 100 breathing entities in a 1x1 sealed box.

**Comment by Jared-Fogle** [comment]
> With the new changes a turf will last about 45 minutes, an emergency tank ~13 minutes and a large tank 4.25 hours

This is reasonable to me. :+1:

**Comment by Timberpoes** [comment]
> 
> 
> Just to be sure, can you test these numbers in game? I wanna be sure they're accurate

From my own testing, I've observed the following behaviour through debugging:
The average breath consists of about 0.0043mol (rounded up) of O2.
Roundstart O2 is 22mol on the station turfs.
Deadly O2 is around 16-16.5mol, where the O2 partial pressure starts dropping below the threshold and people start suffocating.
We would thus expect approx (6 / 0.0043) = ~1400 breaths to start succing enough O2 to create a suffocation scenario.

I created a 1x1 GBJ, fixed the air so it had a roundstart mixture and spawned 20 felinids inside the box.

With the fix in #53565 applied, the felinids started suffocating in around 10 minutes.

600 seconds divived by 8 is 75 breaths. Multiply by 20 for each unique mob gives ~1500 total breaths to drain enough O2 to start suffocation. This matches the simulation and testing data.

All of Tivi's maths are derived from this same set of data and same concepts, since we worked on the maths behind this together, as such I can vouch for their accuracy which is supported by my in-game tests.

**Comment by Tlaltecuhtli** [comment]
i would say give emergency tanks like 1 min instead of literally half of the round, making people actually use the large tank outside of miners

changing tanks vol might cause atmos balance concerns (tiny singlecaps?)

**Comment by MrDoomBringer** [comment]
> This would have to be an area where realism gives way very heavily to game design in order to make breathing air actually matter. Our atmos sim has too much likeness to the real world.

yeah, sounds reasonable

> i would say give emergency tanks like 1 min

honestly something like this would be pretty neat. Maybe not THAT tiny, but somewhere in the single-digit minutes might be pretty fun.

**Comment by msgerbs** [comment]
I agree with shrinking emergency tanks, and I also agree with making that a separate PR. Make atmos actually occasionally need to refill or print fresh tanks. A lot of changes have been justified by "requiring interaction" _coughs in standard borg_, this seems to fall very much in the same realm. I'm not sure 1 minute is entirely reasonable, but 2-5 minutes is totally reasonable for an emergency air tank. You aren't meant to be able to just go about your business as normal in somewhere without sufficient air, you're meant to be fixing the problem or getting out.

Of course the question then becomes what of the extended tanks. I presume those should be shrunk as well, but I think it might then make sense to bring back quad emergency tanks or some other tank that's in between the massive full size tanks for EVA and the tiny emergency tanks. Something that would last 15 - 20 minutes, to let atmos techs have time to fix an air issue without necessarily having to go for the hardsuit just because they need a larger tank slot.

**Comment by Ghilker** [comment]
There is one problem tho, and with faster air consumption it will be more noticeable. 
Basically people breathe in only O2, so they remove only oxy from the air mix and expel co2 that will be eventually scrubbed away. Let's say we start with 20% O2 and 80% N2, after one minute we breathed 1% of O2 that will be replaced by the distro vents while the N2 stays the same amount. The problem I see is that we don't replace only O2 but a mixture so that 1% will be replace by only 0.2% of oxygen and the rest is N2. What would now happen after 30 minutes? That air mix will saturate with N2 and people will start dying from oxiloss even with a perfect air system

**Comment by Winterous** [comment]
I think that emergency tanks should  have a smaller volume, and START at max pressure; currently you grab an oxygen tank from a firefighting closet, go to a pump, fill it up and you've got enough oxygen for like twenty minutes or something. It's for emergencies, it should be ready-to-go already.
Extended capacity could have like 50% extra volume, and also start at max pressure, currently they're double.

It feels like there's no real reason for a normally breathing person to last more than 10 minutes on an emergency tank at full pressure. Whenever I'm in a job that can do it, I'll grab myself a yellow E-tank and fill it at a pump, that'll last around 40 minutes, at a guess, that's simply preposterous.

Blue tank is 1 volume, yellow is 2. Both start at around 1k pressure; the max safe pressure for tanks is around 3k. Make them start just below the leaking threshold, and reduce the volume by a similar ratio. So blue would be .3 volume, yellow .5, something thereabouts. And if we're tripling that with this PR, 1 and 1.5.

**Comment by Dennok** [comment]
> There is one problem tho, and with faster air consumption it will be more noticeable.
> Basically people breathe in only O2, so they remove only oxy from the air mix and expel co2 that will be eventually scrubbed away. Let's say we start with 20% O2 and 80% N2, after one minute we breathed 1% of O2 that will be replaced by the distro vents while the N2 stays the same amount. The problem I see is that we don't replace only O2 but a mixture so that 1% will be replace by only 0.2% of oxygen and the rest is N2. What would now happen after 30 minutes? That air mix will saturate with N2 and people will start dying from oxiloss even with a perfect air system

Yes, but you get minor air alarm in area before O2 drops lover than 16%, and major affter. Until you not in maintenance area, or in ghetto.

Maybe code airalarm to automaticaly start to replace air in zone on minor alarm. At least in general areas like halls, and bar.

**Comment by GuyonBroadway** [comment]
So how does this affect mining?

**Comment by Jared-Fogle** [comment]
@GuyonBroadway Large tanks last for 4.25 hours, as per the PR body. Does that answer your question?

**Comment by GuyonBroadway** [comment]
Partly, do large tanks still fit in an explorers suit slot?

**Comment by Dennok** [comment]
> So how does this affect mining?

Near month before i thinked about self filling oxygen tanks, that take oxygen from atmosphere.
Today i see it in Doctor Who.

**Comment by Jared-Fogle** [comment]
@GuyonBroadway This only changes how much volume they have, not item slots. So yes.

**Comment by GuyonBroadway** [comment]
>Yes.
Then neato.


Thinking about it the problem with mini tanks is everyone is breathing low pressure oxygen as opposed to... well regular air.

Alternative suggestion, try replacing the contents of emergency tanks with a standard air mix and see what happens.

**Comment by Electronics** [comment]
> Maybe code airalarm to automaticaly start to replace air in zone on minor alarm. At least in general areas like halls, and bar.

Problem is with _replace air_ is that near scrubbers the pressure can drop dramatically to damaging levels, and vice versa with the vent pumps.

**Comment by Dennok** [comment]
> > Maybe code airalarm to automaticaly start to replace air in zone on minor alarm. At least in general areas like halls, and bar.
> 
> Problem is with _replace air_ is that near scrubbers the pressure can drop dramatically to damaging levels, and vice versa with the vent pumps.

In other way, make scrubbers scrub excess nitrogen when it % over set % in air alarm.

File: code/game/objects/items/tanks/tank_types.dm
Comment (CRITAWAKETS) [inline code/game/objects/items/tanks/tank_types.dm:121]: You do know that plasmaman belt tanks can be used for breathing things other than plasma by other species, right?

Comment (Dennok) [reply code/game/objects/items/tanks/tank_types.dm:121]: Don't burn the office.
Only cargo and engi remember this.

Comment (msgerbs) [reply code/game/objects/items/tanks/tank_types.dm:121]: Most people don't even fill their O2 tanks at all, if somebody really wants to go get a plasmaman tank and fill it with O2 and waste a far more valuable slot (the belt slot) then more power to them I guess?

**Comment by msgerbs** [comment]
> 
> 
> > > Maybe code airalarm to automaticaly start to replace air in zone on minor alarm. At least in general areas like halls, and bar.
> > 
> > 
> > Problem is with _replace air_ is that near scrubbers the pressure can drop dramatically to damaging levels, and vice versa with the vent pumps.
> 
> In other way, make scrubbers scrub excess nitrogen when it % bigger then set % in air alarm.

That is functionally the same thing, with just a slightly lower pressure drop because it's only taking in 79% of the air instead of 100%. Also please don't fuck with my scrubber settings, thanks. I don't think that, even at 4x the air consumption, you're going to see any real need for this. It still takes 6 hours for even a 3x3 room to be depleted of oxygen, and scrubbers already remove CO2 which naturally causes vents to refill air slightly. The only way this would even be a problem is if another gas is dumped in which raises pressure enough to prevent the vents refilling, but then isn't that exactly the sort of thing where we want atmos techs to be useful and have a job to do? I will be absolutely shocked if even the most packed departments have a problem with this on any normal shift length.

**Comment by kopoba** [comment]
i have bad experience with emergency o2 tanks from backpack box. It ends VERY VERY fast. My space experience was broken by sufocation to death.

**Comment by msgerbs** [comment]
> 
> 
> i have bad experience with emergency o2 tanks from backpack box. It ends VERY VERY fast. My space experience was broken by sufocation to death.

There's currently an unrelated bug causing 8x-ish faster breathing. I've seen a couple humans and a plasmaman die from it. There's a PR open to fix it.

**Comment by Moja10** [comment]
This PR should be denied on the basis that Oranges likes it. Also, this PR doesn't affect plasmameme breathing right?

**Comment by Timberpoes** [comment]
> 
> 
> There is one problem tho, and with faster air consumption it will be more noticeable.
> Basically people breathe in only O2, so they remove only oxy from the air mix and expel co2 that will be eventually scrubbed away. Let's say we start with 20% O2 and 80% N2, after one minute we breathed 1% of O2 that will be replaced by the distro vents while the N2 stays the same amount. The problem I see is that we don't replace only O2 but a mixture so that 1% will be replace by only 0.2% of oxygen and the rest is N2. What would now happen after 30 minutes? That air mix will saturate with N2 and people will start dying from oxiloss even with a perfect air system

You know, that may actually give atmos techs a purpose. Air alarms already have a mode to recycle the air in a room and if air is low across the station, change the ratio of N2/O2 in air distro. It also gives traitor atmos techs plausible deniability to mess around with air distro, siphon rooms, recycle air and more.

I think you accidentally sold me on changes that will make atmos techs a more engaged and more important role across the station!

**Comment by msgerbs** [comment]
I completely agree with @Timberpoes . As is right now, if anyone sees a scrubber on siphon they either immediately kill the AI or kill the atmos tech because it's just... not something that's normally needed. Giving a legitimate reason to more often mess with the air mix, air alarms, etc. would be good.

One think that I'm on the fence about is if it would then make sense to allow remote access similar to the APC controller in the CE's office, or if that's going too far in the direction of powercreep and enabling the greatest plasma fire of all time.

**Comment by SkeletalElite** [comment]
I think ideal numbers for tanks would be like 5 minutes on the blue tank and 10 minutes on a yellow ones and have double stack tanks be acquirable somehow and have them last like 20 minutes in case you need higher capacity but can't spare a slot for the big tanks.

**Comment by Timberpoes** [comment]
Do not merge until #53565 or any other PR addressing the fact people currently breathe 4x more often than intended is merged first.

If that problem is solved, this should be fine for merging.

**Review by Timberpoes — APPROVED** [review]

File: code/game/objects/items/tanks/tank_types.dm
Comment (Timberpoes) [reply code/game/objects/items/tanks/tank_types.dm:121]: Do note that on live servers right now, players can get a plasmaman refill tank and use it to get similar behaviour described here. Tank sizes go 1L/2L/6L (plasma belt tank)/8L (double emergency oxygen tank)

After this change, they go 3L/6L/24L (plasma belt tank)/24L (double emergency oxygen tank)

If players wanted bigger internals, they can already do this on live.

Yes, this IS a very minor buff to people using plasma belt tanks as for their internals compared to before but, I think that is a price I'm willing to pay.

**Review by Timberpoes — APPROVED** [review]

**Comment by Timberpoes** [comment]
In the future we can jimmy around with O2 tank capacities and durations. Feel free to hit up a discussion on #coding-general in the official /tg/Station Discord and/or make a forum post if you have more thoughts and ideas on this topic.

Atmos affects pretty much everyone on the station in some way, shape or form and at some point in time. I'm mindful to limit this PR purely to its current scope. It's a small, simple change. If it somehow breaks the entire game and crashes the server in infinite loops, it's easy to undo.

**Comment by msgerbs** [comment]
@Timberpoes Do you think the CL for this is clear enough? It's kind of vague for people who don't read GitHub to understand what was changed.

**Comment by Timberpoes** [comment]
> 
> 
> @Timberpoes Do you think the CL for this is clear enough? It's kind of vague for people who don't read GitHub to understand what was changed.

I have ASSUMED DIRECT CONTROL and edited the CL myself, so I don't keep dragging Tivi back in for minor changes.

**Comment by cacogen** [comment]
Don't nerf emergency oxygen tanks

**Comment by msgerbs** [comment]
> 
> 
> Don't nerf emergency oxygen tanks

They're comically large right now.

**Comment by Timberpoes** [comment]
I'll merge this in about an hour if nobody pipes up any pressing reasons not to now that we're not all hyper-ventilating our way across the station.

**Review by MrsTonedOne — APPROVED** [review]

**Pull request merged**
